src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -1380,35 +1386,61 @@ SDCPUThread::sendToMemory(shared_ptr<InflightInst> inst_ptr,
     inst_ptr->notifyMemorying();
 
     weak_ptr<InflightInst> weak_inst(inst_ptr);
-    auto callback =
-        [this, weak_inst] (Fault fault) {
-            onExecutionCompleted(weak_inst, fault);
-        };
 
-    if (sreq) {
-        assert(sreq->high && sreq->low);
-        if (write) {
+    if (write) {
+        PacketPtr resp = Packet::createWrite(sreq ? sreq->main : req);
+        if (resp->hasData()) {
+            resp->dataStatic(data.get());
+        }
+        resp->makeResponse();
+
+        Fault f = inst_ptr->staticInst()->completeAcc(resp, inst_ptr.get(),
```

Comment:
When does O3 call this? I know that stores "happen" after commits in the O3 CPU. You can see this if you use O3PipeView. (I can show you later, if you'd like.)

Reply:
From further inspection, it actually looks like O3 doesn't call completeAcc() on stores, except in the case of StoreConditionals... funnily enough, we call completeAcc() on all stores, and I haven't made sure to handle the case for StoreConditionals since I was assuming we weren't going to need that case for the paper deadline

Reply:
looking at the exec-ns generated implementations, the store instructions basically all don't have any code in completeAcc except return NoFault;

Reply:
Interesting. My thought with the pipe view is that I want the trace to have the right timestamp for store completion. However, now that I think more about it, pipeview is it's own special trace, which we haven't implemented yet. Ignore this comment :).